core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.ts

```diff
@@ -56,4 +57,21 @@ export class NgbdModalWorkflowExecutionsComponent implements OnInit {
     }
     return "";
   }
+
+  onBookmarkToggle(row: WorkflowExecutionsEntry) {
+    const wasPreviouslyBookmarked = row.bookmarked;
+
+    // Update bookmark state locally.
+    row.bookmarked = !wasPreviouslyBookmarked;
```

Comment:
Why do you need a local copy and hover actions?

Reply:
I don't want the local bookmark filling to have to be synchronously dependent on the network request. That loading lag could be irritating. So the idea was to just make the change, and then make the network request, and if that fails later, only then do we have an issue.

Why I needed to make my own hover is because the class for fill needed to be dynamically set. CSS hover does not give me the ability to change javascript and angular has no built in hover functionality.